File: app/src/main/java/com/github/willjgriff/ethereumwallet/di/EthereumModule.java
package com.github.willjgriff.ethereumwallet.di;

import android.content.Context;

import com.github.willjgriff.ethereumwallet.data.ethereum.EthereumKeystoreLocation;
import com.github.willjgriff.ethereumwallet.data.ethereum.EthereumAccountManager;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

/**
 * Created by Will on 29/01/2017.
 */

@Module
public class EthereumModule {

	@Provides
	@Singleton
	EthereumKeystoreLocation providesKeystoreLocation(Context context) {
		return new EthereumKeystoreLocation(context);
	}

	@Provides
	@Singleton
	EthereumAccountManager providesEthereum(EthereumKeystoreLocation ethereumKeystoreLocation) {
		return new EthereumAccountManager(ethereumKeystoreLocation);
	}
}

File: app/src/main/kotlin/com/github/wiljgriff/ethereumwallet/di/modules/EthereumModule.kt
package com.github.wiljgriff.ethereumwallet.di.modules

import android.content.Context
import com.github.wiljgriff.ethereumwallet.ethereum.EthereumManager
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

/**
 * Created by williamgriffiths on 17/04/2017.
 */
@Module
class EthereumModule {

    @Provides
    @Singleton
    fun provideEthereumManager(context: Context) =
            EthereumManager(context)

    @Provides
    @Singleton
    fun provideNodeDetails(ethereumManager: EthereumManager) =
            ethereumManager.nodeDetails

    @Provides
    @Singleton
    fun provideWalletAccountManager(ethereumManager: EthereumManager) =
            ethereumManager.walletAccountManager

    @Provides
    @Singleton
    fun provideAccountBalance(ethereumManager: EthereumManager) =
            ethereumManager.accountBalance

}